R/class_mrgmod.R

```diff
@@ -114,6 +114,8 @@ protomod <- list(model=character(0),
                  preclean=FALSE,
                  atol=1E-8,
                  rtol=1E-8,
+                 ss_rtol=1e-6, 
```

Comment:
1e-8 is the dlsoda (ODE) default; I'm keeping that for now; wanted to relax the tolerance a little bit by default on the steady state calculation after talking it over a bit with some people.  Going with 1e-6 for now but negotiable.

Reply:
Does the feedback show 1e-8 for SS rtol too strict?

Reply:
Partly think that ode rtol is too strict as a default; the model **never** actually gets to steady state and we were wondering if the difference between 1e-6 and 1e-8 is just not worth it (some suggestions for 1e-2).  We do get some failures, but the maximum number of iterations is so high that I doubt there are many cases where a model failed with 1e-8 but would succeed with 1e-6.  

Do you think I should just stick with the same default for both?  I'm fine with that (looking to get talked out of 1e-6).

Reply:
> We do get some failures, but the maximum number of iterations is so high that I doubt there are many cases where a model failed with 1e-8 but would succeed with 1e-6.

I doubt that too. A large nb. of iterations could also accumulate too much roundoff error.

In addition, note that the error checking 
https://github.com/metrumresearchgroup/mrgsolve/blob/d13eb94c73e0c0e907e2a95c9043671f22a62949/src/datarecord.cpp#L262

is based on both ssrtol and ssatol. Too much difference between the two values will render ssAtol useless. So I don't think 1e-2 is a good idea, unless we change how tolerance is used in error checking.

Reply:
I just set it back to 1e-8; so no change in behavior.  If we want to drop rtol (I'd like to at some point) we can do it for both.  For now, no change in behavior unless the user opts in.

Reply:
thx. Off the topic do you have any example SS solver fails to converge? I'm happening to evaluate Torsten's SS solver(based on CVODES & KINSOL) and looking for examples that could potentially break it.

Reply:
I'm not totally sure what causes the problems that I've seen.  It's almost always been when there is some randomness in the parameters and you can get some really long half-lives so it takes a long time to get to steady state.  the mrgsolve setup can "fail" because I put a limit on how long I let the search go on.  I'm guessing that is my primary case.  I also thought I saw some problems when the dosing interval was unusually large compared to the absorption rate constant ... so that dosing compartment can get really small.  Not sure why it would cause it to fail but it seemed like something was funny there.  Not sure I've tested this with the new rtol / atol setup that is in there now.  Maybe that would factor in somehow. 

So if I wanted to get this to fail, I'd simulate that upper tail of Vd and ask for a lot of precision on the steady state determination.